actions/upload_sessions.go

```diff
@@ -161,6 +169,19 @@ func (usr *UploadSessionResource) CreateBeta(c buffalo.Context) error {
 		return err
 	}
 
-	res := uploadSessionCreateRes{UploadSession: u, ID: u.ID.String()}
+	res := uploadSessionCreateRes{
+		UploadSession: u,
+		ID:            u.ID.String(),
+		Invoice:       CreateInvoice(req.StorageLengthInYears, req.FileSizeBytes),
+	}
 	return c.Render(200, r.JSON(res))
 }
+
+func CreateInvoice(storageLengthInYears int, fileSizeBytes int) invoice {
```

Comment:
1. Let's make `getInvoice` a method on the `uploadSession` model.
    1. This method would return `totalCost` for Cost and `ethAddrAlpha` for EthAddress
2. `calculatePayment` should be called in the `StartUploadSession` method (which would set its own `totalCost` property)

Reply:
@AaronVasquez   
>This method would return totalCost for Cost and ethAddrAlpha for EthAddress   

It seems like we would want to check the type here (for alpha or beta) and then return the alpha or beta address based on that result?

Reply:
Yeah, the type should already be set by the time this method gets called. Btw, totalCost should be the entire cost, beta will just expect to receive half. I think this will be less confusing by having upload_sessions cost consistent between the two brokers